### ISSUE TITLE
Please consider retries for the Octokit client

### ISSUE THREAD

**Issue by awendt** [actions/github-script#104]
**Is your feature request related to a problem? Please describe.**

I just ran into a problem where a call to the Github API unexpectedly returned a 404 error. This failed our workflow. The error was intermittent and wasn't reproducible when I re-ran the job.

**Describe the solution you'd like**

I'd like any API calls made by the `github` client to be retried, preferably using the [retry plugin](https://github.com/octokit/plugin-retry.js)

**Describe alternatives you've considered**

I've considered adding the retry logic to our workflow but this defeats the purpose of having a pre-authenticated API client.

**Comment by joshmgross** [comment]
Since not all consumers may want this behavior, we could support it behind an optional input

```yaml
  enable-retries:
    description: Whether to use the retry plugin to retry GitHub requests
    default: false
```

**Comment by andris-zalitis** [comment]
@joshmgross we're hitting rate-limiting in our GitHub action. Would definitely use retry plugin if it was available.

**Comment by sarkis** [comment]
I'm not sure retry plugin would help for rate limiting, but similarly would be great to have an option to enable [throttling](https://github.com/octokit/plugin-throttling.js/) plugin, as well.

**Comment by MichaelJJ** [comment]
Would like to see retry capability also

**Comment by AWNCypher** [comment]
👍

**Comment by jsoref** [comment]
See also #229

**Comment by awendt** [comment]
@markmssd Thanks for this! Looks like this would solve our problem. I'd be fine with closing this then.

**Comment by joshmgross** [comment]
👋 This is now available in `v6.3.0`, thanks @luketomlinson for the implementation 🙇 

See https://github.com/actions/github-script/tree/v6.3.0#retries for more information

I'll bump the `v6` to point to `v6.2.0` tomorrow.
